db/migrations/sdp-migrations/2024-09-04.0-alter-receivers-table-indexes.sql

```diff
@@ -0,0 +1,34 @@
+-- +migrate Up
+-- Remove existing unique constraint on phone_number
+ALTER TABLE receivers DROP CONSTRAINT IF EXISTS payments_account_phone_number_key;
+
+DROP INDEX IF EXISTS payments_account_phone_number_key;
+
+-- Make phone_number nullable
+ALTER TABLE receivers ALTER COLUMN phone_number DROP NOT NULL;
+
+-- Add check constraint to ensure at least one contact method is provided
+ALTER TABLE receivers ADD CONSTRAINT receiver_contact_check
+    CHECK (phone_number IS NOT NULL OR email IS NOT NULL);
+
+-- Create unique indexes on both phone_number and email separately
```

Comment:
I had to change to two separate indexes instead of one composite index. The reason is that we don't want the following scenario: 
1. Insert `Receiver1` with `Email1` 
2. Insert `Receiver2` with `SMS2`
3. Update `Receiver2` with `Email1`. 
Now we have two receivers with `Email1`, subsequent disbursements to `email1` would cause two receivers to be matched to one instruction. This doesn't result in double spend because of a check we have in place that makes sure the number of instructions matches the numbers of receivers, but we should still make sure this doesn't happen.